<launch>
  <!-- usb camera -->
  <rosparam ns="usb_cam" command="load" file="$(find manicapture)/cfg/camctl.yaml" />
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
  </node>

  <!-- below is apriltag stuff -->
  <arg name="launch_prefix" default="" />

  <!-- configure camera input -->
  <arg name="camera_name" default="/usb_cam" />
  <arg name="image_topic" default="image_raw" />
  <arg name="queue_size" default="1" />

  <!-- apriltag_ros continuous detection node -->
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros_continuous_node" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)">
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />

    <param name="publish_tag_detections_image" type="bool" value="true" /><!-- default: false -->
    <param name="queue_size" type="int" value="$(arg queue_size)" />

    <!-- load parameters (incl. tag family, tags, etc.) -->
    <rosparam command="load" file="$(find manicapture)/cfg//tagctl.yaml"/>
  </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find manicapture)/launch/manipose.rviz" />

</launch>
